### PR TITLE
fix(examples): fix jest example on windows

### LINKED ISSUE #1454: Jest Example fails on Windows
# 🐞 bug report

### Affected Rule

jest_test

### Is this a regression?

no

### Description

The jest_test rule fails on windows with the following error:

```
Error: Can't find a root directory while resolving a config file path.
Provided path to resolve: ./jest.config.js
```

## 🔬 Minimal Reproduction

https://github.com/bazelbuild/rules_nodejs/tree/master/examples/jest

running `bazel test ...`

## 🔥 Exception or Error

see above in explanation

## 🌍  Your Environment

**Operating System:**

Windows 10

**Output of `bazel version`:**

<pre>
  <code>
Bazelisk version: v1.1.0
Build label: 1.2.1
Build target: bazel-out/x64_windows-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Nov 26 15:24:17 2019 (1574781857)
Build timestamp: 1574781857
Build timestamp as int: 1574781857
  </code>
</pre>

**Rules version (SHA):**

<pre>
  <code>
a54b2511d6dae42c1f7cdaeb08144ee2808193a088004fc3b464a04583d5aa2e
  </code>
</pre>

### PR DESCRIPTION
**Pull request by lukasholzer**
Fixes #1454

image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by alexeagle** [comment]
Failing on CI

```

INFO: From Testing //:test:
--
  | ==================== Test output for //:test:
  | Error: Can't find a root directory while resolving a config file path.
  | Provided path to resolve: ./jest.config.js
  | cwd: C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\bazel-out\x64_windows-fastbuild\bin\test.bat.runfiles\examples_jest
  | at _default (C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-config\build\resolveConfigPath.js:69:11)
  | at readConfig (C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-config\build\index.js:152:49)
  | at readConfigs (C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-config\build\index.js:373:26)
  | at C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-cli\node_modules\@jest\core\build\cli\index.js:155:58
  | at Generator.next (<anonymous>)
  | at asyncGeneratorStep (C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-cli\node_modules\@jest\core\build\cli\index.js:108:24)

```
do you know how to fix?

File: examples/jest/.bazelrc [hunk 1]
```diff
@@ -1 +1,6 @@
 import %workspace%/../../common.bazelrc
+
+# Do not build runfile forests by default. If an execution strategy relies on runfile
```
Comment (alexeagle) [inline examples/jest/.bazelrc:3]: Interesting, thanks for the links! It seems like this isn't particular to the jest example, right? Maybe we should have this in the /common.bazelrc so it gets picked up by all our tests, as well as new workspaces created with `@bazel/create` ?

File: examples/jest/BUILD.bazel [hunk 2]
```diff
@@ -11,10 +11,6 @@ jest_test(
     ] + glob(["__snapshots__/*.snap"]),
     jest_config = ":jest.config.js",
     tags = [
-        # Need to set the pwd to avoid jest needing a runfiles helper
-        # Windows users with permissions can use --enable_runfiles
-        # to make this test work
-        "no-bazelci-windows",
```
Comment (alexeagle) [inline examples/jest/BUILD.bazel:17]: nice thanks for removing one of these

File: examples/jest/.bazelrc (hunk 1 above)
Comment (lukasholzer) [reply examples/jest/.bazelrc:3]: yea! that is a good Idea this will also help normal run files (nodejs_binary) to work on windows.

**Comment by lukasholzer** [comment]
> Failing on CI
> 
> ```
> 
> INFO: From Testing //:test:
> --
>   | ==================== Test output for //:test:
>   | Error: Can't find a root directory while resolving a config file path.
>   | Provided path to resolve: ./jest.config.js
>   | cwd: C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\bazel-out\x64_windows-fastbuild\bin\test.bat.runfiles\examples_jest
>   | at _default (C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-config\build\resolveConfigPath.js:69:11)
>   | at readConfig (C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-config\build\index.js:152:49)
>   | at readConfigs (C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-config\build\index.js:373:26)
>   | at C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-cli\node_modules\@jest\core\build\cli\index.js:155:58
>   | at Generator.next (<anonymous>)
>   | at asyncGeneratorStep (C:\users\b\_bazel_b\o3fgohdy\execroot\examples_jest\node_modules\jest-cli\node_modules\@jest\core\build\cli\index.js:108:24)
> ```
> 
> do you know how to fix?

I'm currently on it – seems that the runfilesHelper has a bug on windows as well (need to debug the wall of rc flags that are used on the ci :D).

**Comment by lukasholzer** [comment]
@alexeagle The updated regex in the `launcher.sh` fixes all issues where this error message occured: `ERROR: Manifest file c:\Users\ADMINI~1\_BAZEL~1\rgyky3dz\execroot\EXAMPL~1\bazel-out\x64_windows-fastbuild\bin\test.bat.runfiles\examples_jest\test.bat.runfiles_manifest does not exist.`

The problem is that the msys2 package sometimes writes paths with a lower letter: `c:\tmp\qwm2tovj\execroot\dynatr~1\bazel...`. This results in going into the if and appending the current working directory so the `BAZEL_NODE_PATCH_REQUIRE` had always a wrong path on windows. Wich results in that some nodejs_tests where crashing.

File: examples/jest/.bazelrc (hunk 1 above)
Comment (lukasholzer) [reply examples/jest/.bazelrc]: done. Moved it into the common bazelrc

**Comment by lukasholzer** [comment]
@alexeagle can you help me with the failing ci – suddenly the whole ci seems to fail. No clue why

**Comment by alexeagle** [comment]
sorry I haven't had time to repro, maybe someone in Slack #javascript can help?

**Review by alexeagle — APPROVED** [review]

**Comment by alexeagle** [comment]
hey @lukasholzer I tried to green this up, but still failing on windows:

```

>>>> FAIL: The node binary 'nodejs_linux_amd64/bin/nodejs/bin/node' not found in runfiles.
--
  | This node toolchain was chosen based on your uname ' x86_64'.
  | Please file an issue to https://github.com/bazelbuild/rules_nodejs/issues if
  | you would like to add your platform to the supported rules_nodejs node platforms. <<<<

```

could you TAL?

**Comment by lukasholzer** [comment]
@alexeagle I get now a different error locally when running: `bazel run //examples:examples_jest` 


image

**Review by alexeagle — APPROVED** [review]
thanks!

**Pull request merged**